packages/framework/src/Model/Order/Watcher/TransportAndPaymentWatcher.php

```diff
@@ -102,20 +108,24 @@ protected function checkTransportPrice(
         Transport $transport,
         Currency $currency,
         OrderPreview $orderPreview,
-        $domainId
-    ) {
+        int $domainId
+    ): bool {
         $transportPrices = $this->getRememberedTransportPrices();
 
         if (array_key_exists($transport->getId(), $transportPrices)) {
             $rememberedTransportPriceValue = $transportPrices[$transport->getId()];
+            if (!($rememberedTransportPriceValue instanceof Money)) {
```

Comment:
this is for some backwards-compatibility? It would deserve maybe a PHP comment

Reply:
As same as PaymentPrice below

Reply:
Or recommend to delete sessions after upgrading to `7.0.0` because there maybe could be more session related BC-breaks

Reply:
Yes, it was meant to deal with BC. Maybe we should add a comment
.

Reply:
Added in da672c060

Reply:
So... you think we should keep this backwards compatibility? I don't know why - we are releasing new major version that can contain BC-breaks.

Reply:
The upgrade may be hard as it is, we wanted to avoid an extra unnecessary issue that may be hard to prevent without affecting users (clearing the sessions completely).

Reply:
I would appreciate kind of `@deprecated` or `TODO THIS CAN BE REMOVED IN... ` comment